pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -174,13 +174,44 @@ func AddToManager(mgr manager.Manager, r *ReconcileClusterPool, concurrentReconc
 	}
 
 	// Watch for changes to ClusterDeploymentCustomizations
-	if err := c.Watch(&source.Kind{Type: &hivev1.ClusterDeploymentCustomization{}}, &handler.EnqueueRequestForObject{}); err != nil {
+	if err := c.Watch(
+		&source.Kind{Type: &hivev1.ClusterDeploymentCustomization{}},
+		handler.EnqueueRequestsFromMapFunc(
+			requestsForCDCResources(r.Client, r.logger)),
+	); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func requestsForCDCResources(c client.Client, logger log.FieldLogger) handler.MapFunc {
```

Comment:
Thanks, this looks right ✓

Reply:
Looks like this is now filtering down to only pools containing the CDC in their inventory. That should be fine :+1: